Diplom_3

Chaplin-86 opened the pull request:
Написаны тесты на проверку следующей функциональности:
Регистрация
Вход 
Переход в личный кабинет 
Переход из личного кабинета в конструктор 
Выход из аккаунта
Переходы к разделам конструктора

Chaplin-86 committed:
Отредактирован pom.xml;
Добавлены классы PageObject (MainPage, LogInPage, ProfilePage, SignUpPage, ResetPasswordPage);
В классы PageObject добавлены локаторы и методы;
Добавлен пакет Api с классами и методами для создания пользователя;
Добавлены классы DriverRule, EnvConfig

Chaplin-86 committed:
Отредактирован pom.xml;
Добавлены тестовые классы ConstructorTests, LogInTests, PersonalProfileTests;
Отредактирован код в классе DriverRule;
Добавлены отчеты Allure для тестов в Chrome и Yandex

Chaplin-86 committed:
Отредактирован pom.xml;
Добавлены тестовые классы ConstructorTests, LogInTests, PersonalProfileTests;
Отредактирован код в классе DriverRule;
Добавлены отчеты Allure для тестов в Chrome и Yandex

Chaplin-86 committed:
Отредактирован pom.xml;
Добавлены тестовые классы ConstructorTests, LogInTests, PersonalProfileTests;
Отредактирован код в классе DriverRule;
Добавлены отчеты Allure для тестов в Chrome и Yandex

Chaplin-86 committed:
Добавлена аннотация '@Step' в классы пакета pageobject;
Добавлен тестовый класс SignUpTests;

Chaplin-86 committed:
добавлена папка 'target'

mkouznetsov-git commented on src/test/java/ConstructorTests.java:
```diff
@@ -0,0 +1,60 @@
+    @Test
+    @DisplayName("Проверка перехода к разделу 'Булки'")
+    public void goToBunTabTest() {
+        WebDriver driver = driverRule.getDriver();
```
Метод getDriver() статичный, обращаться к нему следует по имени класса, а не объекта. Это надо поправить во всех классах

mkouznetsov-git commented on src/test/java/ConstructorTests.java:
```diff
@@ -0,0 +1,60 @@
+        driver.get(BASE_URL);
+
+        new MainPage(driver)
+                .clickSauceTab();
```
Если удалить клик по табу, то тест всё-равно проходит успешно. Значит, проверка перехода по табам не работает.

mkouznetsov-git commented on src/test/java/SignUpTests.java:
```diff
@@ -0,0 +1,89 @@
+
+    @Test
+    @DisplayName("Проверка успешной регистрации")
+    public void checkSuccessfulRegistrationTest() {
```
Этот тест стабильно падает. Сможешь разобраться, почему?

mkouznetsov-git replied:
Не исправлено.

Chaplin-86 committed:
Исправлены локаторы для элементов buns, sauce, filling в классе MainPage;
Удалены лишние ожидания из методов нажатия на вкладки "Булки", "Соусы", "Начинки";
Отлажены тесты в классе ConstructorTests

mkouznetsov-git commented on src/test/java/ConstructorTests.java:
```diff
@@ -0,0 +1,61 @@
+import static praktikum.EnvConfig.BASE_URL;
+
+public class ConstructorTests {
+    private DriverRule driverRule = new DriverRule();
```
Объект driverRule лишний, он нигде не используется. Удаляй его везде.

Chaplin-86 committed:
Удалены лишние поля в классе ConstructorTests

Chaplin-86 committed:
Исправлен тест на проверку успешной регистрации в класс SignUpTests

mkouznetsov-git commented on src/test/java/SignUpTests.java:
```diff
@@ -0,0 +1,75 @@
+import api.User;
+import api.UserClient;
```
Лишний импорт

mkouznetsov-git commented on src/test/java/SignUpTests.java:
```diff
@@ -0,0 +1,75 @@
+import io.qameta.allure.Description;
+import io.qameta.allure.junit4.DisplayName;
+import io.restassured.RestAssured;
+import io.restassured.response.ValidatableResponse;
```
Лишний импорт

Chaplin-86 committed:
Удален лишний импорт в классе SignUpTests

Chaplin-86 committed:
Изменила передачу токена в методе удаления клиента
